docs/docs/concepts/key-parameters.md

```diff
@@ -0,0 +1,85 @@
+# Key Parameters in Desired State Configuration
+
+PowerShell Desired State Configurations (DSC) are used to configure a system in a desired state. This is done by comparing the current state of the system with the desired state and applying the necessary changes to get to the desired state. The state is defined by the configuration. Every part of the configuration needs to be unique. Uniqueness is defined by the keys of the resource. The keys are the parameters that are used to identify a resource. If the keys are not unique, the configuration will not compile.
+
+This is a unique configuration for two users in a tenant:
+
+```powershell
+
+Configuration MyConfig {
+
+    Import-DscResource -ModuleName Microsoft365DSC
+
+    Node localhost {
+
+        # Create a new user 1
+        AADUser User1 {
+            UserPrincipalName = "User1@tenant.onmicrosoft.com"
+            DisplayName       = "User"
+        }
+
+        # Create a new user 2
+        AADUser User2 {
+            UserPrincipalName = "User2@tenant.onmicrosoft.com"
+            DisplayName       = "User"
+        }
+    }
+}
+
+```
+
+Whereas this configuration is not unique and will not compile:
+
+```powershell
+
+Configuration MyConfig {
+
+    Import-DscResource -ModuleName Microsoft365DSC
+
+    Node localhost {
+
+        # Create a new user 1
+        AADUser User1 {
+            UserPrincipalName = "User1@tenant.onmicrosoft.com"
+            DisplayName       = "User1"
+        }
+
+        # Create a new user 2 - with the same UPN as user 1
+        AADUser User2 {
+            UserPrincipalName = "User1@tenant.onmicrosoft.com"
+            DisplayName       = "User2"
+        }
+    }
+}
+
+```
+
+There are two resources with the same key combination. User1 and User2 do have the same UserPrincipalName and would configure different DisplayNames. This is not allowed and will not compile. The same is true for other resources. For example, if you would have two AADGroups with the same UserPrincipalName, the configuration would not compile.
```

Comment:
In the above examples we only have one Key parameter. Shouldn't we add the comment that multiple parameters can be marked key and that the combination must be unique within a configuration?

Reply:
done

Reply:
done